pkg/apis/rook.io/v1/types.go

```diff
@@ -116,12 +119,17 @@ type LabelsSpec map[KeyType]Labels
 type Labels map[string]string
 
 type StorageClassDeviceSet struct {
-	Name                 string                     `json:"name,omitempty"`                 // A unique identifier for the set
-	Count                int                        `json:"count,omitempty"`                // Number of devices in this set
-	Resources            v1.ResourceRequirements    `json:"resources,omitempty"`            // Requests/limits for the devices
-	Placement            Placement                  `json:"placement,omitempty"`            // Placement constraints for the device daemons
-	PreparePlacement     *Placement                 `json:"preparePlacement,omitempty"`     // Placement constraints for the device preparation
-	Config               map[string]string          `json:"config,omitempty"`               // Provider-specific device configuration
+	Name  string `json:"name,omitempty"`  // A unique identifier for the set
+	Count int    `json:"count,omitempty"` // Number of devices in this set
+	// +kubebuilder:pruning:PreserveUnknownFields
+	Resources v1.ResourceRequirements `json:"resources,omitempty"` // Requests/limits for the devices
+	// +kubebuilder:pruning:PreserveUnknownFields
+	Placement Placement `json:"placement,omitempty"` // Placement constraints for the device daemons
+	// +kubebuilder:pruning:PreserveUnknownFields
+	PreparePlacement *Placement `json:"preparePlacement,omitempty"` // Placement constraints for the device preparation
+	// +kubebuilder:pruning:PreserveUnknownFields
+	Config map[string]string `json:"config,omitempty"` // Provider-specific device configuration
+	// +kubebuilder:pruning:PreserveUnknownFields
```

Comment:
Why do we need preserveunknownfields for a type that comes from k8s? Does it require that?